docs/source/_ext/ghfiles.py

```diff
@@ -0,0 +1,65 @@
+import os.path as path
+import subprocess
+import shlex
+from sphinx.util import logging
+from docutils import nodes
+logger = logging.getLogger(__name__)
+
+
+# use an old git trick, to get the top-level, could have used ../ etc.. but
+# this will be fine..
+top = subprocess.check_output(shlex.split(
+    "git rev-parse --show-toplevel")).strip().decode("utf-8")
+
+
+def make_ref(text):
+    """ Make hyperlink to Github """
+    full_path = path.join(top, text)
+    if path.isfile(full_path):
+        ref = "https://github.com/numba/numba/blob/master/" + text
+    elif path.isdir(full_path):
+        ref = "https://www.github.com/numba/numba/tree/master/" + text
+    else:
+        logger.warn("Failed to find file in repomap: " + text)
+        ref = "https://www.github.com/numba/numba"
+    return ref
+
+
+def intersperse(lst, item):
+    """ Insert item between each item in lst. """
+    result = [item] * (len(lst) * 2 - 1)
+    result[0::2] = lst
+    return result
+
+
+def ghfile_role(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    """ Emit hyperlink nodes for a given file in repomap. """
+    my_nodes = []
+    if "{" in text:  # myfile.{c, h} - make two nodes
```

Comment:
Super minor nit, but the comment is misleading, as this code only works when there is no space between the extension options: `myfile.{c,h}`, which is the case in the docs.  The only reason I note it here is that it made me worried I had put spaces in the docs below, so I had to go check.  :)

Reply:
Minor nits are the best nits!